curvefs/src/mds/metaserverclient/metaserver_client.cpp

```diff
@@ -189,27 +227,62 @@ FSStatusCode MetaserverClient::CreatePartition(uint32_t fsId, uint32_t poolId,
     partition->set_end(idEnd);
     partition->set_txid(0);
 
-    if (channel_.Init(addr.c_str(), nullptr) != 0) {
-        LOG(ERROR) << "Init channel to metaserver: " << addr
-                   << " failed!";
-        return FSStatusCode::RPC_ERROR;
-    }
+    do {
+        if (refreshLeader) {
+            auto ret = GetLeader(poolId, copysetId, addrs, &leader);
+            if (ret != FSStatusCode::OK) {
+                LOG(ERROR) << "CreatePartition fail, get leader fail"
+                           << ", poolId = " << poolId
+                           << ", copysetId = " << copysetId;
+                return ret;
+            }
+            if (channel_.Init(leader.c_str(), nullptr) != 0) {
+                LOG(ERROR) << "Init channel to metaserver: " << leader
+                           << " failed!";
+                return FSStatusCode::RPC_ERROR;
+            }
+        }
+        cntl.Reset();
+        cntl.set_timeout_ms(options_.rpcTimeoutMs);
+        MetaServerService_Stub stub(&channel_);
+        stub.CreatePartition(&cntl, &request, &response, nullptr);
```

Comment:
almost same with previous CreateRootInode, you can follow https://github.com/opencurve/curve/blob/a0b92d3680c5125bb66f8ad1a6deb33624446288/src/tools/mds_client.cpp#L1063 to do some improvement.

Reply:
> almost same with previous CreateRootInode, you can follow
> 
> https://github.com/opencurve/curve/blob/a0b92d3680c5125bb66f8ad1a6deb33624446288/src/tools/mds_client.cpp#L1063
> 
> to do some improvement.

fixed